### PR TITLE
Improve integration test coverage + fix any bugs found

### PR DESCRIPTION
**Pull request by neketka**
### Summary 

This pull request improves integration test coverage across the backend and attempts to fix any bugs found during this process. It also fixes the dbreset script to run exclusively inside docker and not be affected by any external postgres instances. It also normalizes start.sh line endings to LF so that docker does not complain on some git setups on windows.

- [x] Dockerized dbreset
- [x] Automated CRLF to LF conversion for start.sh
- [x] Uncommented previously failing organization tests and fixed related bugs
- [x] Added a complete playthrough test
- [x] Implemented correct per-field permission checking (extended CASL's functionality)
- [x] Disabled setCurrentChallenge since this is not currently applicable
- [x] Added default event and challenge creation 
- [x] Added timestamping (createdAt, updatedAt)
- [x] Fixed incorrect event tracker id being reported to sendProtected

### Test Plan 

The integration tests included are the test plan.

### CODE REVIEW

**Review by cathli66 — COMMENTED** [review]
It's difficult to understand a lot of the CASL permissions code but nothing appears glaringly wrong to me. I checked over the new e2e tests and they look good! They all pass on my end, so I'm convinced that the CASL works fine. Would appreciate it if you could address my comments before I approve!

File: server/src/challenge/challenge.service.ts [hunk 1]
```diff
@@ -124,15 +125,14 @@ export class ChallengeService {
...
+      data: { score: { increment: curChallenge.points } },
     });
 
     await this.prisma.eventTracker.update({
       where: { id: eventTracker.id },
       data: {
-        score: { increment: 1 },
+        score: { increment: curChallenge.points },
         curChallenge: { connect: { id: nextChallenge.id } },
-        completedChallenges: { connect: { id: prevChal.id } },
```
[the remark this answers is not in the document]
Comment (neketka) [reply server/src/challenge/challenge.service.ts:135]: We do, this was a duplication of the line above so I removed it

File: scripts/setup.js [hunk 2]
```diff
@@ -12,10 +16,13 @@ cd ../game
 flutter pub get
 cd ..
 npm run dbreset
+npm run tests:e2e
 */
 
 async function main() {
-  console.log("Checking if docker is installed (will fail if not)");
+  console.log("Converting start.sh to LF (required with some Git setups)");
+  execSync("npx crlf --set=LF ./server/start.sh");
```
[the remark this answers is not in the document]
Comment (neketka) [reply scripts/setup.js]: Moved to docker file

**Review by cathli66 — APPROVED** [review]
Addressed all my comments, thanks! Also tested authorization thoroughly on admin frontend - everything works as expected. All integration tests also pass and look correct to me.

**Pull request merged**